### PR TITLE
lxc exec: Fix exit codes for signaled processes 

### PR DESCRIPTION
**Pull request by hamistao**
With this, processes spawned by `lxc exec` stopped due to being signaled now return the proper exit code to LXD.

From the [docs](https://pkg.go.dev/os#ProcessState.Sys):

> Sys returns system-dependent exit information about the process. Convert it to the appropriate underlying type, such as [syscall.WaitStatus](https://pkg.go.dev/syscall#WaitStatus) on Unix, to access its contents.

### CODE REVIEW

File: shared/util_linux.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 	"path/filepath"
 	"reflect"
 	"strings"
+	"syscall"
```
Comment (tomponline) [inline shared/util_linux.go:16]: > NOTE: Most of the functions, types, and constants defined in this package are also available in the [golang.org/x/sys](https://pkg.go.dev/golang.org/x/sys) package. That package has more system call support than this one, and most new code should prefer that package where possible. See https://golang.org/s/go1.4-syscall for more information.

Should we use [golang.org/x/sys](https://pkg.go.dev/golang.org/x/sys) instead?

Comment (hamistao) [reply shared/util_linux.go:16]: Casting will only work if using `syscall.WaitStatus`

**Comment by hamistao** [comment]
@tomponline Tests are updated as discussed, no more clean stop

**Comment by tomponline** [comment]
> @tomponline Tests are updated as discussed, no more clean stop

Looks like ur still importing "syscall# through

File: shared/util_linux.go (hunk 1 above)
Comment (tomponline) [reply shared/util_linux.go:16]: Oh i missed this comment, ok ta

**Review by tomponline — APPROVED** [review]

**Pull request merged**